mbq/client/contrib/permissions.py

```diff
@@ -227,32 +243,51 @@ def _cache_write(self, doc: CachedPermissionsDoc) -> None:
             except Exception as e:
                 raise ServerError("Error writing to cache") from e
 
-    def _has_permission(self, person_id: UUIDType, scope: str, spec: RefSpec) -> bool:
+    def _has_permission(
+        self, person_id: UUIDType, scope: str, specs: List[RefSpec]
+    ) -> bool:
+        """Returns bool of whether the given person has the given
+        scope on ALL RefSpecs specified.
+        """
         person_id = str(person_id)
 
-        cached_doc = self._cache_read(person_id, [spec])
+        cached_doc = self._cache_read(person_id, specs)
 
         if not cached_doc:
-            if isinstance(spec.ref, int):
-                assert spec.type is not None
-                fetched_doc = self.os_core_client.fetch_permissions_for_location(
-                    person_id, spec.ref, spec.type
-                )
+            if len(specs) > 1 or specs[0].ref == "global":
+                logger.debug("Using fetch_all_permissions")
+                fetched_doc = self.os_core_client.fetch_all_permissions(person_id)
             else:
-                fetched_doc = self.os_core_client.fetch_permissions(person_id, spec.ref)
+                spec = specs[0]
+                if isinstance(spec.ref, int):
+                    assert spec.type is not None
+                    logger.debug("Using fetch_permissions_for_location")
+                    fetched_doc = self.os_core_client.fetch_permissions_for_location(
+                        person_id, spec.ref, spec.type
+                    )
+                else:
+                    logger.debug("Using fetch_permissions")
```

Comment:
this message might be a bit better if you say that you're fetching by ref

Reply:
Intent is to show which part of the protocol I'm accessing. I can write my own logging statements in the OS Core client but I can't control the internal client from here